examples/customwebhookbot_django.py

```diff
@@ -0,0 +1,198 @@
+    text = (
+        f"To check if the bot is still running, call <code>{url}/healthcheck</code>.\n\n"
+        f"To post a custom update, call <code>{payload_url}</code>."
+    )
+    await update.message.reply_html(text=text)
+
+
+async def webhook_update(update: WebhookUpdate, context: CustomContext) -> None:
+    """Handle custom updates."""
+    chat_member = await context.bot.get_chat_member(chat_id=update.user_id, user_id=update.user_id)
+    payloads = context.user_data.setdefault("payloads", [])
+    payloads.append(update.payload)
+    combined_payloads = "</code>\n• <code>".join(payloads)
+    text = (
+        f"The user {chat_member.user.mention_html()} has sent a new payload. "
+        f"So far they have sent the following payloads: \n\n• <code>{combined_payloads}</code>"
+    )
+    await context.bot.send_message(
+        chat_id=context.bot_data["admin_chat_id"], text=text, parse_mode=ParseMode.HTML
+    )
+
+
+async def telegram(request: HttpRequest) -> HttpResponse:
+    """Handle incoming Telegram updates by putting them into the `update_queue`"""
+    await ptb_application.update_queue.put(
+        Update.de_json(data=json.loads(request.body), bot=ptb_application.bot)
+    )
+    return HttpResponse()
+
+
+async def custom_updates(request: HttpRequest) -> HttpResponse:
+    """
+    Handle incoming webhook updates by also putting them into the `update_queue` if
+    the required parameters were passed correctly.
+    """
+    try:
+        user_id = int(request.GET["user_id"])
+        payload = request.GET["payload"]
+    except KeyError:
+        return HttpResponseBadRequest(
+            "Please pass both `user_id` and `payload` as query parameters.",
+        )
+    except ValueError:
+        return HttpResponseBadRequest("The `user_id` must be a string!")
+
+    await ptb_application.update_queue.put(WebhookUpdate(user_id=user_id, payload=payload))
+    return HttpResponse()
+
+
+async def health(_: HttpRequest) -> HttpResponse:
+    """For the health endpoint, reply with a simple plain text message."""
+    return HttpResponse("The bot is still running fine :)")
+
+
+# Set up PTB application and a web application for handling the incoming requests.
+
+context_types = ContextTypes(context=CustomContext)
+# Here we set updater to None because we want our custom webhook server to handle the updates
+# and hence we don't need an Updater instance
+ptb_application = (
+    Application.builder().token("TOKEN").updater(None).context_types(context_types).build()
+)
+
+# register handlers
+ptb_application.add_handler(CommandHandler("start", start))
+ptb_application.add_handler(TypeHandler(type=WebhookUpdate, callback=webhook_update))
+
+urlpatterns = [
+    path("telegram", telegram, name="Telegram updates"),
+    path("submitpayload", custom_updates, name="custom updates"),
+    path("healthcheck", health, name="health check"),
+]
+settings.configure(ROOT_URLCONF=__name__, SECRET_KEY=uuid4().hex)
```

Comment:
why is this outside `main()`?

Reply:
b/c djange treats the file as o module and needs to be able to import `urlpatterns` from it …